packages/apps/human-app/server/src/integrations/exchange-oracle/exchange-oracle.gateway.ts

```diff
@@ -171,4 +175,34 @@ export class ExchangeOracleGateway {
     };
     return this.callExternalHttpUtilRequest<JobsDiscoveryResponse>(options);
   }
+
+  async registerWorker(command: RegisterWorkerCommand) {
+    const isRegistrationNeeded =
+      await this.kvStoreGateway.getExchangeOracleRegistrationNeeded(
+        command.oracleAddress,
+      );
+
+    if (!isRegistrationNeeded) {
+      throw new HttpException('Invalid request', 400);
+    }
```

Comment:
This should be done in workerService.registerWorker.
Gateway should be used just for outgoing requests